src/lib/libast/misc/signal.c

```diff
@@ -60,8 +60,8 @@
 #define	sa_mask		sv_mask
 #endif
 
-extern Sig_handler_t
-signal(int sig, Sig_handler_t fun)
+Sig_handler_t
+_ast_signal(int sig, Sig_handler_t fun)
```

Comment:
I don't think this rename is correct. The `#define signal _ast_signal` in map.c is emitted conditionally. If `signal` is *not* redefined to `_ast_signal`, then this rename should break things. If `signal` *is* redefined to `_ast_signal`, then this rename is unnecessary because the function will be implicitly renamed by the macro expansion. I could be missing something but this is how I think it works.

Oh, wait, I see the '#define signal' and '#undef signal' at the top. It should be possible to delete these (as well as undoing the rename) if that logic is transferred to map.c -- i.e., simply change line 138 in map.c to: `#if !__HAIKU__`.

I could be missing something and unfortunately I still don't have access to my x86_64 VMs right now, so I hope you have the time to test this.

Reply:
The latest changes I applied have been successfully tested on Linux, Haiku, *BSD, illumos and Cygwin. (The `!__HAIKU` block was deleted as it's no longer necessary with the `_ast_signal` definition in map.c.)